tests/aequilibrae/project/test_field_editor.py

```diff
@@ -1,137 +1,101 @@
-
-            # Then we add a new field to the table
-            val2 = self.randomword(10)
-            curr.execute(f'Alter table "{table}" add column "{val2}" NUMERIC;')
-            curr.execute(f"pragma table_info({table})")
-            fields = [x[1] for x in curr.fetchall() if x[1] == val2]
-            self.assertEqual([val2], fields, "failed to add a new field")
-
-            table = FieldEditor(self.proj, table)
-            self.proj.conn.commit()
-            curr = self.proj.conn.cursor()
-            curr.execute(f'Select count(*) from "attributes_documentation" where attribute="{val}"')
-            self.assertEqual(curr.fetchone()[0], 0, f"clean the table on loading failed {val}")
-
-            curr.execute(f'Select count(*) from "attributes_documentation" where attribute="{val2}"')
-            self.assertEqual(curr.fetchone()[0], 1, "clean the table on loading failed")
-            print(table._original_values[val2])
-
-            self.proj.conn.commit()
-            del curr
-
-    def test_save(self):
-        for tab in ["modes", "links", "nodes", "link_types"]:
-            table = FieldEditor(self.proj, tab)
-            random_val = self.randomword(30)
-            if "alpha" in table._original_values.keys():
-                table.alpha = random_val
-                table.save()
-                table2 = FieldEditor(self.proj, tab)
-
-                self.assertEqual(table2.alpha, random_val, "Did not save values properly")
-
-            if "link_id" in table._original_values.keys():
-                table.link_id = random_val
-                table.save()
-                table2 = FieldEditor(self.proj, tab)
-
-                self.assertEqual(table2.link_id, random_val, "Did not save values properly")
-
-            if "node_id" in table._original_values.keys():
-                table.node_id = random_val
-                table.save()
-                table2 = FieldEditor(self.proj, tab)
-
-                self.assertEqual(table2.node_id, random_val, "Did not save values properly")
-
-            self.proj.conn.commit()
+from pathlib import Path
+
+import pytest
+from aequilibrae.project.field_editor import FieldEditor
+
+
+class TestFieldEditor:
+    @pytest.fixture(scope="session")
+    def project_session(self, create_empty_project_session):
+        return create_empty_project_session()
+
+    @pytest.fixture(scope="session")
+    def database_backup(self, project_session):
+        return Path(project_session.project_base_path).joinpath("project_database.sqlite").read_bytes()
+
+    @pytest.fixture
+    def cleanup_database(self, project_session, database_backup):
+        """An optional fixture that can be used to revert the project's database in case the test
+        modifies it. This adds some extra overhead to the test (teardown)
+        """
+        yield
+        Path(project_session.project_base_path).joinpath("project_database.sqlite").write_bytes(database_backup)
```

Comment:
Same with these... they seem quite generic and useful in other tests.

Reply:
and I get what you are saying here - these cleanup scripts are "subsets of a full wipe and rebuild" that make sense if we know we are just testing "field editor".